DSM/HE/HE_DSM_tindex.py

```diff
@@ -0,0 +1,336 @@
+            tries = 100
+            pass
+        except URLError as e:
+            if tries == 14:
+                grass.message("URLError !!!")
+                grass.message(url_today)
+                grass.fatal(e)
+            grass.message("RETRYING: " + url_today)
+            tries += 1
+            sleep(15)
+            pass
+        except Exception as e:
+            if tries == MAX_RETRIES:
+                grass.message("Exception !!!")
+                grass.message(url_today)
+                grass.fatal(e)
+                pass
+            grass.message("RETRYING: " + url_today)
+            tries += 1
+            sleep(15)
+    if test_url == 200:
+        return True
+    else:
+        # grass.message(_(f"{url_today} is not reachable."))
+        return False
+
+
+def create_urls(type, krs, gem):
+    """Create possible urls for KRS data"""
+    gem_vars = generate_gem_var(gem)
+    type_vars = [type]
+    if type == "Kreisfreie Stadt":
+        type_vars.append("Stadt")
+    urls = []
+    for type_var in type_vars:
+        for gem_var in gem_vars:
+            urls.extend(
+                [
+                    requote_uri(f"{BASE_URL}/{type_var} {krs}/{gem_var} - DOM1.zip"),
+                    requote_uri(f"{BASE_URL}/{type_var} {krs.split(' ')[0]}/{gem_var} - DOM1.zip"),
+                    requote_uri(f"{BASE_URL}/{krs}/{gem_var} - DOM1.zip"),
+                    requote_uri(f"{BASE_URL}/{krs}/{gem_var.split(' ')[0]} - DOM1.zip"),
+                ]
+            )
+    return urls
+
+
+def generate_gem_var(gem):
+    """Generate a list with variants of the gem because of abbreviations"""
+    gem_vars = [gem]
+    if "v.d." in gem:
+        gem_vars.append(gem.replace("v.d.", " v. d. ").replace("  ", " "))
+        gem_vars.append(gem.replace("v.d.", " von der ").replace("  ", " "))
+        gem_vars.append(gem.replace("v.d.", " vor der ").replace("  ", " "))
+    if "i.Odw." in gem:
+        gem_vars.append(gem.replace("i.Odw.", "im Odenwald"))
+        gem_vars.append(gem.replace("i.Odw.", "i. Odw"))
+    if "a.d." in gem:
+        gem_vars.append(gem.replace("a.d.", " a. d. ").replace("  ", " "))
+        gem_vars.append(gem.replace("a.d.", " a.d. ").replace("  ", " "))
+        gem_vars.append(gem.replace("a.d.", " an der ").replace("  ", " "))
+    if "a." in gem:
+        gem_vars.append(gem.replace("a.", " a. ").replace("  ", " "))
+        gem_vars.append(gem.replace("a.", " am ").replace("  ", " "))
+    if gem in ["Michelbuch"]:
+        gem_vars.append(f"Gemarkung {gem} (gemeindefrei)")
+    if gem == "Erbach":
+        gem_vars.append("Erbach (Odenwald)")
+    if gem == "Neukirchen":
+        gem_vars.append("Neukirchen (Knüllgebirge)")
+    if gem in ["Lorch"]:
+        gem_vars.append(f"{gem} am Rhein")
+    if gem == "Heppenheim":
+        gem_vars.append(f"{gem} (Bergstraße)")
+    return gem_vars
+
+
+"""MAIN PART"""
+
+os.chdir("/src/tile-indices/DSM/HE/")
+# os.chdir("DSM/HE/")
```

Comment:
The commented path, would be as within this repo. If adjustable, maybe also move to top